web/src/portico/landing-page.ts

```diff
@@ -8,100 +8,117 @@ import iphone_image from "../../images/app-screenshots/zulip-iphone-rough.png";
 import {page_params} from "../page_params";
 
 import {detect_user_os} from "./tabbed-instructions";
+import type {UserOS} from "./tabbed-instructions";
 import render_tabs from "./team";
 
-export function path_parts() {
+export function path_parts(): string[] {
     return window.location.pathname.split("/").filter((chunk) => chunk !== "");
 }
 
-const apps_events = function () {
-    const info = {
+type ZulipOsSpecificInfo = Record<
+    UserOS,
+    {
+        alt: "Android" | "iOS" | "Linux" | "macOS" | "Windows";
+        app_store_link?: string;
+        app_type: "mobile" | "desktop";
+        description: string;
+        download_link?: string;
+        image: string;
+        install_guide?: string;
+        mac_arm64_link?: string;
+        play_store_link?: string;
+        show_instructions: boolean;
+    }
+>;
+
+const apps_events = function (): void {
+    const info: ZulipOsSpecificInfo = {
         windows: {
-            image: microsoft_image,
             alt: "Windows",
+            app_type: "desktop",
             description:
                 "Zulip for Windows is even better than Zulip on the web, with a cleaner look, tray integration, native notifications, and support for multiple Zulip accounts.",
             download_link: "/apps/download/windows",
-            show_instructions: true,
+            image: microsoft_image,
             install_guide: "/help/desktop-app-install-guide",
-            app_type: "desktop",
+            show_instructions: true,
```

Comment:
If you're going to sort these parameters, please do it in its own commit, separate from the other changes; it makes it way harder to read to have these reorderings squashed in with the actual TypeScript logic.

(But it's possible this is just an editor setting you should turn off?)

Reply:
Yea, I'll revert and make a new commit for it.